x-pack/plugins/triggers_actions_ui/public/application/sections/action_connector_form/create_connector_flyout/index.test.tsx

```diff
@@ -469,6 +469,35 @@ describe('CreateConnectorFlyout', () => {
         name: 'My test',
         secrets: {},
       });
+      expect(queryByTestId('required-field-error-label')).not.toBeInTheDocument();
+    });
+
+    it('show Required all fields message', async () => {
+      const { getByTestId } = appMockRenderer.render(
+        <CreateConnectorFlyout
+          actionTypeRegistry={actionTypeRegistry}
+          onClose={onClose}
+          onConnectorCreated={onConnectorCreated}
+          onTestConnector={onTestConnector}
+        />
+      );
+      await act(() => Promise.resolve());
+
+      await userEvent.click(getByTestId(`${actionTypeModel.id}-card`));
+
+      await waitFor(() => {
+        expect(getByTestId('test-connector-text-field')).toBeInTheDocument();
+      });
+
+      await userEvent.type(getByTestId('test-connector-text-field'), 'My text field', {
+        delay: 100,
+      });
+
+      await userEvent.click(getByTestId('create-connector-flyout-save-btn'));
+
+      expect(onClose).not.toHaveBeenCalled();
+      expect(onConnectorCreated).not.toHaveBeenCalled();
+      expect(getByTestId('required-field-error-label')).toBeInTheDocument();
```

Comment:
```suggestion
    it('show Required all fields message', async () => {
      appMockRenderer.render(
        <CreateConnectorFlyout
          actionTypeRegistry={actionTypeRegistry}
          onClose={onClose}
          onConnectorCreated={onConnectorCreated}
          onTestConnector={onTestConnector}
        />
      );

      await userEvent.click(await screen.findByTestId(`${actionTypeModel.id}-card`));

      expect(await screen.findByTestId('test-connector-text-field')).toBeInTheDocument();

      await userEvent.type(
        await screen.findByTestId('test-connector-text-field'),
        'My text field',
        {
          delay: 100,
        }
      );

      await userEvent.click(await screen.findByTestId('create-connector-flyout-save-btn'));

      expect(onClose).not.toHaveBeenCalled();
      expect(onConnectorCreated).not.toHaveBeenCalled();
      expect(await screen.findByTestId('required-field-error-label')).toBeInTheDocument();
```

The other tests in this file are very outdated(I assume you used those as reference).

`screen` is preferred over the results of `render`, `findBy*` is preferred over `getBy*`, and `await act(() => Promise.resolve());` is not needed(and potentially problematic) when `waitFor` and `findBy` should be used to wait for pending promises to be resolved.

Reply:
Could you please also make sure that in this test all fields are required(and the message shows)? And maybe add another where when **not all** fields are required the message is not displayed? 🙏

Reply:
Thank you for the suggestion. I will update these tests accordingly.